tests/test_cache.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 Northwestern University.
+#
+# invenio-analytics-importer is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+import json
+
+from invenio_analytics_importer.cache import generate_cache
+
+
+def write_json(filepath, content):
+    """Write json filepath."""
+    with open(filepath, "w") as f:
+        json.dump(content, f)
+    return filepath
+
+
+def test_generate_cache(running_app, record_factory, tmp_path):
```

Comment:
Are these arguments passed by the testing configuration? In particular I was curious where `record_factory` comes from.

Reply:
Yup, like @sharpattack  mentioned those are coming from https://github.com/galter-dmt/invenio-analytics-importer/blob/master/tests/conftest.py . Shamefully those are fixtures repeated across a lot of projects by this point, so they are not very interesting / should be reused somehow. `running_app` is just a running InvenioRDM application with lots of preloaded data necessary to be able to create records and make searches. `record_factory` is to create records. `tmp_path` is a [pytest provided fixture](https://docs.pytest.org/en/stable/how-to/tmp_path.html) to cleanly handle temporary diretory set up and tear down.